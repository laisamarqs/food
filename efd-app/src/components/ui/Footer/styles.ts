import styled from 'styled-components'
import { breakpoints, colors } from '../../../styles'
import { Logo } from '../../../styles'

export const FooterContainer = styled.footer`
  background-color: ${colors.primaryLight};
  text-align: center;
  padding: 2.5rem 0;
  bottom: 0;
  height: 300px;

  ${Logo} {
    margin-bottom: 2rem;
  }

  p {
    color: ${colors.primaryDark};
    font-size: 0.625rem;

    @media only screen and (max-width: ${breakpoints.mobile}) {
      padding: 0 2rem;
    }
  }
`

export const SocialMediaContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom: 5rem;

  a {
    cursor: pointer;
  }

  img {
    width: 48px;
    height: 48px;
  }
`