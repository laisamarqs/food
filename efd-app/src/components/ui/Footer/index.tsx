import { Link } from 'react-router-dom'

import * as Style from './styles'
import { Logo } from '../../../styles'

import logoImage from '../../../assets/images/logo.svg'
import instagramIcon from '../../../assets/images/instagram-icon.svg'
import facebookIcon from '../../../assets/images/facebook-icon.svg'
import twitterIcon from '../../../assets/images/twitter-icon.svg'

const Footer = () => (
  <Style.FooterContainer>
    <Link to="/" title="Home">
      <Logo src={logoImage} alt="EFOOD" />
    </Link>
    <Style.SocialMediaContainer>
      <a href="#" title="Instagram">
        <img src={instagramIcon} alt="Icone do instagram" />
      </a>
      <a href="#" title="Facebook">
        <img src={facebookIcon} alt="Icone do facebook" />
      </a>
      <a href="#" title="Twitter">
        <img src={twitterIcon} alt="Icone do twitter" />
      </a>
    </Style.SocialMediaContainer>
    <p>
      A efood é uma plataforma para divulgação de estabelecimentos
      responsabilidade pela entrega, qualidade <br /> dos produtos é toda do
      estabelecimento contratado.{' '}
    </p>
  </Style.FooterContainer>
)

export default Footer