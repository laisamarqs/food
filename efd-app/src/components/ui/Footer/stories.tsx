import type { Meta } from '@storybook/react'

import Footer from '.'

const meta = {
  title: 'ui/Footer',
  component: Footer
} satisfies Meta<typeof Footer>

// eslint-disable-next-line react-refresh/only-export-components
export default meta

export const Template = () => {
  return <Footer />
}