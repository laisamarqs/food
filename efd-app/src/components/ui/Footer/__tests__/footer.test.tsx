import '@testing-library/jest-dom'
import { MemoryRouter } from 'react-router-dom'
import { render, screen } from '@testing-library/react'
import Footer from '..'
import { expect } from 'vitest'

describe('<Footer />', () => {
  it('should render correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    )
    expect(screen.getByAltText('EFOOD'))

    expect(container.firstChild).toMatchSnapshot()
  })
})