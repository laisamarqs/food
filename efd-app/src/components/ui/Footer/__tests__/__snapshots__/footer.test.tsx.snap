// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Footer /> > should render correctly 1`] = `
<footer
  class="sc-dcJsrY faKNNM"
>
  <a
    href="/"
    title="Home"
  >
    <img
      alt="EFOOD"
      class="sc-aXZVg drWPxd"
      src="/src/assets/images/logo.svg"
    />
  </a>
  <div
    class="sc-iGgWBj lkGQIz"
  >
    <a
      href="#"
      title="Instagram"
    >
      <img
        alt="Icone do instagram"
        src="/src/assets/images/instagram-icon.svg"
      />
    </a>
    <a
      href="#"
      title="Facebook"
    >
      <img
        alt="Icone do facebook"
        src="/src/assets/images/facebook-icon.svg"
      />
    </a>
    <a
      href="#"
      title="Twitter"
    >
      <img
        alt="Icone do twitter"
        src="/src/assets/images/twitter-icon.svg"
      />
    </a>
  </div>
  <p>
    A efood é uma plataforma para divulgação de estabelecimentos responsabilidade pela entrega, qualidade 
    <br />
     dos produtos é toda do estabelecimento contratado.
     
  </p>
</footer>
`;